packages/expo-image-manipulator/ios/ImageManipulations.swift

```diff
@@ -0,0 +1,211 @@
+}
+
+/**
+ Makes sure the image is oriented up and not mirrored.
+ Guarantees that the original pixel data matches the displayed orientation.
+ */
+internal func fixImageOrientation(_ image: UIImage) throws -> UIImage {
+  guard let cgImage = image.cgImage else {
+    throw ImageNotFoundError()
+  }
+  guard let colorSpace = cgImage.colorSpace else {
+    // That should never happen as `colorSpace` is empty only when the image is a mask.
+    throw ImageColorSpaceNotFoundError()
+  }
+
+  var transform = CGAffineTransform.identity
+
+  switch image.imageOrientation {
+  case .down, .downMirrored:
+    transform = transform.translatedBy(x: image.size.width, y: image.size.height)
+    transform = transform.rotated(by: Double.pi)
+  case .left, .leftMirrored:
+    transform = transform.translatedBy(x: image.size.width, y: 0)
+    transform = transform.rotated(by: Double.pi / 2)
+  case .right, .rightMirrored:
+    transform = transform.translatedBy(x: 0, y: image.size.height)
+    transform = transform.rotated(by: -Double.pi / 2)
+  default:
+    break
+  }
+
+  switch image.imageOrientation {
+  case .upMirrored, .downMirrored:
+    transform = transform.translatedBy(x: image.size.width, y: 0)
+    transform = transform.scaledBy(x: -1, y: 1)
+  case .leftMirrored, .rightMirrored:
+    transform = transform.translatedBy(x: image.size.height, y: 0)
+    transform = transform.scaledBy(x: -1, y: 1)
+  default:
+    break
+  }
+
+  let context = CGContext(
+    data: nil,
+    width: Int(image.size.width),
+    height: Int(image.size.height),
+    bitsPerComponent: cgImage.bitsPerComponent,
+    bytesPerRow: 0,
+    space: colorSpace,
+    bitmapInfo: CGImageAlphaInfo.premultipliedLast.rawValue
+  )
+
+  guard let context = context else {
+    throw ImageContextLostError()
+  }
+
+  context.concatenate(transform)
+
+  switch image.imageOrientation {
+  case .left, .leftMirrored, .right, .rightMirrored:
+    context.draw(cgImage, in: CGRect(x: 0, y: 0, width: image.size.height, height: image.size.width))
+  default:
+    context.draw(cgImage, in: CGRect(x: 0, y: 0, width: image.size.width, height: image.size.height))
+    break
+  }
+
+  guard let newCGImage = context.makeImage() else {
+    throw ImageDrawingFailedError()
+  }
+  return UIImage(cgImage: newCGImage)
+}
+
+/**
+ Helper function for drawing the image in graphics context.
+ Throws appropriate errors when the context is missing or the image couldn't be rendered.
+ */
+fileprivate func drawInNewContext(size: CGSize, drawing: (CGContext) -> ()) throws -> UIImage {
+  UIGraphicsBeginImageContext(size)
+
+  guard let context = UIGraphicsGetCurrentContext() else {
+    UIGraphicsEndImageContext()
+    throw ImageContextLostError()
+  }
+
+  drawing(context)
+
+  guard let newImage = UIGraphicsGetImageFromCurrentImageContext() else {
+    UIGraphicsEndImageContext()
+    throw NoImageInContextError()
+  }
+
+  UIGraphicsEndImageContext()
+  return newImage
+}
+
```

Comment:
**⚠️  SwiftLint Violation: Trailing Newline**
Files should have a single trailing newline.